src/Symfony/Component/Form/Tests/Extension/Core/EventListener/FixUrlProtocolListenerTest.php

```diff
@@ -20,38 +20,38 @@
 
 class FixUrlProtocolListenerTest extends TestCase
 {
-    public function testFixHttpUrl()
+    public function provideUrlToFix()
     {
-        $data = 'www.symfony.com';
-        $form = new Form(new FormConfigBuilder('name', null, new EventDispatcher()));
-        $event = new FormEvent($form, $data);
-
-        $filter = new FixUrlProtocolListener('http');
-        $filter->onSubmit($event);
-
-        $this->assertEquals('http://www.symfony.com', $event->getData());
+        return [
+            ['www.symfony.com'],
+            ['twitter.com/@symfony'],
```

Comment:
please also add a test using the `@` in the query string (which is why you have `?` in the regex), and maybe we should also account for things like `www.symfony.com#foo@bar` by adding `#` in the range.

Reply:
Added `#` and test cases.
Also added `localhost` in tests that should prefixed (I was tempted to make `.` mandatory in host part).